pkg/server/admin.go

```diff
@@ -2441,4 +2463,33 @@ func (s *adminServer) hasAdminRole(ctx context.Context, sessionUser string) (boo
 	return bool(dbDatum), nil
 }
 
-var errInsufficientPrivilege = status.Error(codes.PermissionDenied, "this operation requires admin privilege")
+func (s *adminServer) hasRoleOption(
+	ctx context.Context, user string, roleOption roleoption.Option,
+) (bool, error) {
+	if user == security.RootUser {
+		// Shortcut.
+		return true, nil
+	}
+	rows, cols, err := s.server.sqlServer.internalExecutor.QueryWithCols(
```

Comment:
Is it possible to share these queries with the planner, maybe helper methods that take in an ie interface?